### PR TITLE
Fix psalm and deprecations

### PR DESCRIPTION
**Pull request by VincentLanglet**
Pedantic

### CODE REVIEW

File: src/Admin/Pool.php [hunk 1]
```diff
@@ -269,28 +269,26 @@ public function hasGroup($group)
      */
     public function getDashboardGroups()
```
Comment (VincentLanglet) [inline src/Admin/Pool.php:270]: I refactor the method to help psalm to understand the return type.

File: src/Admin/Pool.php [hunk 2]
```diff
@@ -343,9 +341,8 @@ public function getAdminsByGroup($group)
      *
      * @return AdminInterface|null
      *
-     * @phpstan-template T of object
-     * @phpstan-param class-string<T> $class
-     * @phpstan-return AdminInterface<T>|null
+     * @phpstan-param class-string $class
+     * @phpstan-return AdminInterface|null
```
Comment (VincentLanglet) [inline src/Admin/Pool.php:345]: We can't be sure that the Admin use our class as the subject, since we could pass anything as `$adminClasses` in the constructor.

File: src/Twig/GlobalVariables.php [hunk 3]
```diff
@@ -61,6 +61,7 @@ public function __construct($adminPool, ?string $mosaicBackground = null)
                 Pool::class
             ), \E_USER_DEPRECATED);
 
+            /** @psalm-suppress UndefinedMethod */
```
Comment (VincentLanglet) [inline src/Twig/GlobalVariables.php:64]: This block is deprecated and will be removed, no need to pass time on it.

File: composer.json [hunk 4]
```diff
@@ -85,7 +85,7 @@
         "symfony/browser-kit": "^4.4 || ^5.1",
         "symfony/css-selector": "^4.4 || ^5.1",
         "symfony/filesystem": "^4.4 || ^5.1",
-        "symfony/maker-bundle": "^1.25",
+        "symfony/maker-bundle": "^1.29",
```
Comment (greg0ire) [inline composer.json:88]: What does this have to do with Psalm or deprecations?

Comment (VincentLanglet) [reply composer.json]: Removed

**Review by OskarStark — APPROVED** [review]
